### PR TITLE
Allow full Number range for chooseFloat

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

**Comment by jacereda** [comment]
Somewhat related, shouldn't there be a `Bounded`-like type-class where `bottom`/`top` are finite (`Limited`?), so that something like `chooseFloat lower upper` could be used?

**Comment by hdgarrood** [comment]
I’m not really a fan of a bounded-like type class where bottom/top are finite. What laws would it satisfy? Would it make sense for any types other than Number?

File: test/Main.purs [hunk 1]
```diff
@@ -84,6 +86,8 @@ main = do
   quickCheck     $ 4 >?  3
   quickCheckFail $ 4 <=? 3
 
+  randomSample (MGen.chooseFloat ((-1.7976931348623157e+308)) (1.7976931348623157e+308)) >>= assert <<< all isFinite
```
Comment (hdgarrood) [inline test/Main.purs]: Can you add a log statement just before this line so that if this fails, it will be a little clearer as to why?

**Comment by JordanMartinez** [comment]
This looks good to me. Just curious but why wasn't this PR merged previously?

**Review by JordanMartinez — APPROVED** [review]

**Pull request merged**